gfx/gfx_widgets.c

```diff
@@ -1748,10 +1742,11 @@ void gfx_widgets_frame(void *data)
    }
 
    /* Draw all messages */
-   for (i = 0; i < p_dispwidget->current_msgs->size; i++)
+   SLOCK_LOCK(p_dispwidget->current_msgs_lock);
+
+   for (i = 0; i < p_dispwidget->current_msgs_size; i++)
```

Comment:
In `gfx_widgets_iterate()` you read `p_dispwidget->current_msgs_size` *before* locking `p_dispwidget->current_msgs_lock` - this seems quite safe in that context

But here, you read `p_dispwidget->current_msgs_size` *after* locking `p_dispwidget->current_msgs_lock`. Now given that we check for `msg != NULL` in the loop, wouldn't it be safe to cache `p_dispwidget->current_msgs_size` first (without locking), and then only lock and loop if the cached value is `> 0`? This would save us from having to lock a mutex when there are no messages, and thus remove almost all the general performance overheads added by this PR.